### PR TITLE
MoMMI Spaghetti fix

### LINKED ISSUE #11248: MoMMi spaghetti not working properly
(WEB REPORT BY: rattletrap REMOTE: 172.93.109.202:7777)

> Revision c8d679e1bc0bdd3f99107dfa51401c224d0d0650
> 
> General description of the issue
> Eating MoMMi spaghetti only seems to work half the time, sometimes it doesn't make you start beeping booping at all, and other times it gives you the wrong virus and you become a cyborg instead of a MoMMi.
> 
> What you expected to happen
> MoMMi spaghetti turns you into a MoMMi
> 
> What actually happened
> MoMMi spaghetti didn't work after eating it once, second time it worked except it turned me into a cyborg
> 
> Steps to reproduce if possible
> Eat MoMMi spaghetti

### PR DESCRIPTION
**Pull request by coldcola**
here we go, one more time!

this fixes a problem where mommi spaghetti would create cyborgs.

also removes a space from the end of mommi spaghetti's name to clean up formatting.

this closes #11248

### CODE REVIEW

**Review by PJB3005 — APPROVED** [review]

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 1]
```diff
@@ -2423,11 +2424,14 @@
 	if((prob(10) && method == TOUCH) || method == INGEST)
 		M.contract_disease(new /datum/disease/robotic_transformation(0), 1)
```
Comment (clusterfack) [inline code/modules/reagents/Chemistry-Reagents.dm]: Just make this disease typepath a variable that differs between the two, and delete the reaction mob proc on the mommi version

Comment (coldcola) [reply code/modules/reagents/Chemistry-Reagents.dm]: I had that recommended to me and I couldn't for the life of me get that working without fucking the roburger up.

Comment (clusterfack) [inline code/modules/reagents/Chemistry-Reagents.dm]: It's just three lines
```var/diseasetype = /datum/disease/robotic_transformation```
Change ```new /datum/disease/robotic_transformation``` to ```new diseasetype```
and ```diseasetype = /datum/disease/robotic_transformation/mommi``` for mommis

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 2]
```diff
@@ -3008,7 +3008,7 @@
 		if( src.open )
 			return
 
-		var/t = copytext(sanitize(input("Enter what you want to add to the tag:", "Write", null, null) as text|null), 1, MAX_MESSAGE_LEN)
+		var/t = input("Enter what you want to add to the tag:", "Write", null, null) as text
```
Comment (ghost) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: Isn't the sanity there for a reason?

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 3]
```diff
@@ -2408,32 +2408,27 @@
...
-	name = "Autist nanites"
-	id = AUTISTNANITES
+/datum/reagent/nanites
+    name = "Nanites"
+    id = NANITES
+    description = "Microscopic construction robots."
+    reagent_state = LIQUID
+    color = "#535E66" //rgb: 83, 94, 102
+    var/diseasetype = /datum/disease/robotic_transformation
+/datum/reagent/nanites/reaction_mob(var/mob/living/M, var/method = TOUCH, var/volume, var/robot_type)
```
Comment (clusterfack) [inline code/modules/reagents/Chemistry-Reagents.dm]: Remove this "robot_type" var

File: code/modules/reagents/reagent_containers/food/snacks.dm (hunk 2 above)
Comment (coldcola) [reply code/modules/reagents/reagent_containers/food/snacks.dm]: I actually didn't touch that

Comment (coldcola) [reply code/modules/reagents/reagent_containers/food/snacks.dm]: ah OK, so put it back then?

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 4]
```diff
@@ -2408,32 +2408,27 @@
...
 
-/datum/reagent/nanites
-	name = "Nanites"
-	id = NANITES
-	description = "Microscopic construction robots."
-	reagent_state = LIQUID
-	color = "#535E66" //rgb: 83, 94, 102
 
+
+/datum/reagent/nanites
```
Comment (clusterfack) [inline code/modules/reagents/Chemistry-Reagents.dm]: Convert these all back to tabs, you changed them to spaces by copy pasting from discord

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 5]
```diff
@@ -2521,7 +2516,7 @@
 					for(var/datum/disease/D in M.viruses)
 						D.spread = "Remissive"
 						D.stage--
-						if(D.stage < 1)
+						if(D.stage >= 1)
```
Comment (clusterfack) [inline code/modules/reagents/Chemistry-Reagents.dm]: I'm not sure if this makes sense? It's supposed to keep decreasing stage by 1 until its below 1 in which case it cures it

**Pull request merged**